ksqldb-functional-tests/src/test/resources/sql-tests/test.sql

```diff
@@ -104,3 +108,91 @@ ASSERT VALUES bar (rowtime, id, col1) VALUES (2, 2, 2);
+CREATE STREAM foo (id INT KEY, col1 INT) WITH (kafka_topic='foo', value_format='JSON');
+CREATE STREAM bar AS SELECT * FROM foo;
+
+ASSERT STREAM bar (id INT KEY, col1 INT) WITH (kafka_topic='BAZ', value_format='JSON');
+
+----------------------------------------------------------------------------------------------------
+--@test: assert stream with wrong topic format should fail
+
+--@expected.error: io.confluent.ksql.util.KsqlException
+--@expected.message: Expected value format does not match actual for source BAR
+----------------------------------------------------------------------------------------------------
+CREATE STREAM foo (id INT KEY, col1 INT) WITH (kafka_topic='foo', value_format='JSON');
+CREATE STREAM bar AS SELECT * FROM foo;
+
+ASSERT STREAM bar (id INT KEY, col1 INT) WITH (kafka_topic='BAR', value_format='AVRO');
+
+----------------------------------------------------------------------------------------------------
+--@test: assert stream with wrong timestamp column
+
+--@expected.error: io.confluent.ksql.util.KsqlException
+--@expected.message: Expected timestamp column does not match actual for source BAR.
+----------------------------------------------------------------------------------------------------
+CREATE STREAM foo (id INT KEY, col1 BIGINT, col2 BIGINT) WITH (kafka_topic='foo', value_format='JSON');
+CREATE STREAM bar WITH(timestamp='col1') AS SELECT * FROM foo;
+
+ASSERT STREAM bar (id INT KEY, col1 BIGINT, col2 BIGINT) WITH (kafka_topic='BAR', value_format='JSON', timestamp='col2');
+
+----------------------------------------------------------------------------------------------------
+--@test: assert stream with wrong timestamp format
+
+--@expected.error: io.confluent.ksql.util.KsqlException
+--@expected.message: Expected timestamp format does not match actual for source BAR.
+----------------------------------------------------------------------------------------------------
+CREATE STREAM foo (id INT KEY, col1 VARCHAR) WITH (kafka_topic='foo', value_format='JSON');
+CREATE STREAM bar WITH(timestamp='col1', timestamp_format='yyyy') AS SELECT * FROM foo;
+
+ASSERT STREAM bar (id INT KEY, col1 VARCHAR) WITH (kafka_topic='BAR', value_format='JSON', timestamp='col1', timestamp_format='mm');
+
+----------------------------------------------------------------------------------------------------
+--@test: assert stream with wrong topic should fail
+
+--@expected.error: io.confluent.ksql.util.KsqlException
+--@expected.message: Expected serde options does not match actual for source BAR
+----------------------------------------------------------------------------------------------------
+CREATE STREAM foo (id INT KEY, col1 INT) WITH (kafka_topic='foo', value_format='JSON');
+CREATE STREAM bar AS SELECT * FROM foo;
+
+ASSERT STREAM bar (id INT KEY, col1 INT) WITH (kafka_topic='BAR', value_format='JSON', wrap_single_value=false);
```

Comment:
Hummm.... if the assert here needs to have the default value for `wrap_single_value`, then what happens when we add a new property to the WITH clause?  Will all our own and users test with asserts now fail?  This isn't ideal...

Reply:
I'm not sure I understand your concern. The assert here does not need to have the default value of `wrap_single_values` (which is `true`). This is testing for the non-default use case.

Any new properties added to the WITH clause that maintain backwards compatibility will have a default, and therefore should be supported by this syntax.

Reply:
Ah, cool! Ignore me then :D